script/grant_revoke_gem_authority.rb

```diff
@@ -1,166 +1,142 @@
+  ORGANIZATION_NAMES = ['samvera', 'samvera-labs', 'samvera-deprecated']
+else
+  ORGANIZATION_NAMES = ['samvera', 'samvera-labs']
+end
+
+# Some GitHub repositories are named differently from their gems. We
+# could read each repositories .gemspec to determine its
+# name. However, this is a far quicker short-cut.
 KNOWN_MISMATCHED_GEM_NAMES = {
   'active_fedora' => 'active-fedora',
   'fcrepo-admin' => 'fcrepo_admin',
   'questioning_authority' => 'qa'
 }
+
 # GitHub repositories with matching gems that aren't from Samvera
 FALSE_POSITIVES = [
   'hypatia',
   'rdf-vocab',
   'lerna',
-  'hydrangea',
-  'valkyrie'
+  'hydrangea'
 ]
+
 # Gems that do not have their own GitHub repositories
 HANGERS_ON = [
   'hydra-core',
   'hydra-access-controls',
   'sufia-models',
   'curation_concerns-models'
 ]
-# Email addresses that are known not to be registered at rubygems.org
-SKIP_EMAILS = [
-  'geisler@stanford.edu',
-  'dlrueda@stanford.edu',
-  'jgreben@stanford.edu',
-  'lmcglohon@gmail.com',
-  'ssklar@stanford.edu',
-  'tony.zanella@gmail.com',
-  'blalbritton@gmail.com',
-  'dgcliff@northeastern.edu',
-  'jessica@ucsd.edu'
-]
+
 VERBOSE = ENV.fetch('VERBOSE', false)
 
-puts "(Hang in there! This script takes a couple minutes to run.)"
+# See https://guides.rubygems.org/rubygems-org-api/#rate-limits I'm
+# opting for the slower 0.2 so as to not come close to the rate limit.
+GEM_SLEEP_INTERVAL = 0.2
 
-github = Github.new(oauth_token: AUTHORIZATION_TOKEN, auto_pagination: true)
+$stdout.puts "Hang in there! This script takes a couple minutes to run."
 
-# Get the ID of the GitHub-provided "Owners" team from the samvera org
-# We don't currently grant gem ownership to the folks in the other two orgs
-# (This just preserves the prior behavior.)
-owner_team_id = github.orgs.teams.list(org: 'samvera').select { |team| team.name == 'Admins' }.first.id
-owners = github.orgs.teams.list_members(owner_team_id)
-# Start with the prior (known to work) list of email addresses
-committer_map = KNOWN_COMMITTER_EMAIL_ADDRESSES.dup
-owners.each do |owner|
-  user = github.users.get(user: owner.login)
-  # Move along if the user doesn't have an email addy or if there's already an entry in the map
-  next if !user.respond_to?(:email) || user.email.nil? || user.email.empty? || !committer_map[user.login].nil?
-  committer_map[user.login] = user.email
-end
-committer_emails = committer_map.values.sort.uniq
+github = Github.new(oauth_token: AUTHORIZATION_TOKEN, auto_pagination: true)
 
 # Keep track of things
 @errors = []
 @bogus_gem_names = []
 @gem_names = HANGERS_ON
 
-def exists?(name)
-  system("gem owner #{name} > /dev/null 2>&1")
-end
-
-def replace_known_mismatch(name)
-  KNOWN_MISMATCHED_GEM_NAMES.fetch(name, name)
-end
+@definitive_committers = File.read(
+  File.expand_path("../../data/definitive-rubygems-authors-list.txt", __FILE__)
+).split("\n").map { |line| line.sub(/ *#.*$/, "") }.select { |line| !line.empty? }
```

Comment:
I don't feel strongly about this, but a different way to deal with this is to store the list of gem authors as a simple yaml file, and then the YAML parser will deal with the empty lines and comments for us. Something like:

```yaml
- address1@example.org
- address2@example.com
# ...
```

an advantage of this approach is we can validate the YAML in CI (which I see this project doesn't have... but it could?).